docs/source/custom_resolvers.rst

```diff
@@ -0,0 +1,311 @@
+    >>> # same string literal "${uncached}" => same value
+    >>> assert c.cached_3 == c.cached_3 == 1192
+
+
+Custom interpolations can also receive the following special parameters:
+
+- ``_parent_``: the parent node of an interpolation.
+- ``_root_``: The config root.
+
+This can be achieved by adding the special parameters to the resolver signature.
+Note that special parameters must be defined as named keywords (after the `*`):
+
+In this example, we use ``_parent_`` to implement a sum function that defaults to 0 if the node does not exist.
+(In contrast to the sum we defined earlier where accessing an invalid key, e.g. ``"a_plus_z": ${sum:${a}, ${z}}`` will result in an error).
+
+.. doctest::
+
+    >>> def sum2(a, b, *, _parent_):
+    ...     return _parent_.get(a, 0) + _parent_.get(b, 0)
+    >>> OmegaConf.register_new_resolver("sum2", sum2, use_cache=False)
+    >>> cfg = OmegaConf.create(
+    ...     {
+    ...         "node": {
+    ...             "a": 1,
+    ...             "b": 2,
+    ...             "a_plus_b": "${sum2:a,b}",
+    ...             "a_plus_z": "${sum2:a,z}",
+    ...         },
+    ...     }
+    ... )
+    >>> cfg.node.a_plus_b
+    3
+    >>> cfg.node.a_plus_z
+    1
+
+
+Built-in resolvers
+------------------
+
+.. _oc.env:
+
+oc.env 
+^^^^^^
+
+Access to environment variables is supported using ``oc.env``:
+
+Input YAML file:
+
+.. include:: env_interpolation.yaml
+   :code: yaml
+
+.. doctest::
+
+    >>> conf = OmegaConf.load('source/env_interpolation.yaml')
+    >>> conf.user.name
+    'omry'
+    >>> conf.user.home
+    '/home/omry'
+
+You can specify a default value to use in case the environment variable is not set.
+In such a case, the default value is converted to a string using ``str(default)``, unless it is ``null`` (representing Python ``None``) - in which case ``None`` is returned. 
+
+The following example falls back to default passwords when ``DB_PASSWORD`` is not defined:
+
+.. doctest::
+
+    >>> cfg = OmegaConf.create(
+    ...     {
+    ...         "database": {
+    ...             "password1": "${oc.env:DB_PASSWORD,password}",
+    ...             "password2": "${oc.env:DB_PASSWORD,12345}",
+    ...             "password3": "${oc.env:DB_PASSWORD,null}",
+    ...         },
+    ...     }
+    ... )
+    >>> # default is already a string
+    >>> show(cfg.database.password1)
+    type: str, value: 'password'
+    >>> # default is converted to a string automatically
+    >>> show(cfg.database.password2)
+    type: str, value: '12345'
+    >>> # unless it's None
+    >>> show(cfg.database.password3)
+    type: NoneType, value: None
+
+.. _oc.deprecated:
+
+oc.deprecated
+^^^^^^^^^^^^^
+``oc.deprecated`` enables you to deprecate a config node.
+It takes two parameter:
+- ``key``: An interpolation key representing the new key you are migrating to. This parameter is required.
+- ``message``: A message to use as the warning when the config node is being accessed. The default message is
```

Comment:
```suggestion
- ``message``: An optional message to use as the warning when the config node is being accessed. The default message is
```

Reply:
It's not optional but since it has a default value you don't need to worry about it.

Reply:
It's optional in the sense that you don't need to provide it. But anyway, not a big deal => resolving.